pubsub/subscriptions/create_cloud_storage_subscription.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_cloudstorage_subscription]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// createCloudStorageSubscription creates a Pub/Sub subscription that exports messages to Cloud Storage.
+func createCloudStorageSubscription(w io.Writer, projectID, subID string, topic *pubsub.Topic, bucket string) error {
+	// projectID := "my-project-id"
+	// subID := "my-sub"
+	// topic of type https://godoc.org/cloud.google.com/go/pubsub#Topic
```

Comment:
Passing in a complex type like this seems confusing. Could you supply a topic ID instead and construct or create it in the sample?

Reply:
We could, though this is the current pattern for [other subscription samples](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/pubsub/subscriptions/create.go). I'm inclined to keep it consistent and revisit this once we transition to GAPIC client and redoing the samples then.